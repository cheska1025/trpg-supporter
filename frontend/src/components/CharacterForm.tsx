import { useState, FormEvent } from 'react';

type Props = {
  onSubmit: (payload: { name: string; clazz: string; level: number }) => void | Promise<void>;
  isSubmitting?: boolean;
};

export function CharacterForm({ onSubmit, isSubmitting }: Props) {
  const [name, setName] = useState('');
  const [clazz, setClazz] = useState('');
  const [level, setLevel] = useState<number>(1);

  const handleSubmit = (e: FormEvent) => {
    e.preventDefault();
    onSubmit({ name: name.trim(), clazz: clazz.trim(), level: Number(level) || 1 });
  };

  return (
    <form onSubmit={handleSubmit} style={{ display: 'grid', gap: 8, maxWidth: 320 }}>
      <label>
        Name
        <input
          placeholder="e.g. Alice"
          value={name}
          onChange={(e) => setName(e.target.value)}
        />
      </label>
      <label>
        Class
        <input
          placeholder="e.g. Wizard"
          value={clazz}
          onChange={(e) => setClazz(e.target.value)}
        />
      </label>
      <label>
        Level (default 1)
        <input
          type="number"
          min={1}
          value={level}
          onChange={(e) => setLevel(Number(e.target.value))}
        />
      </label>
      <button type="submit" disabled={isSubmitting}>
        {isSubmitting ? 'Creatingâ€¦' : 'Create'}
      </button>
    </form>
  );
}
