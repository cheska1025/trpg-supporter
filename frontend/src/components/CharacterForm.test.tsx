import { render } from '@testing-library/react';
import { withClient } from '@/test/utils';   // 추가

import { CharacterForm } from './CharacterForm';

describe('CharacterForm', () => {
  it('renders', () => {
    const { container } = render(
      withClient(<CharacterForm onSubmit={async () => {}} />)
    );
    expect(container).toMatchSnapshot();
  });
});
