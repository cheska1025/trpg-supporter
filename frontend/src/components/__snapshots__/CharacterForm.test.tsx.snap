// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CharacterForm > renders 1`] = `
<div>
  <form
    style="display: grid; gap: 8px; max-width: 320px;"
  >
    <label>
      Name
      <input
        placeholder="e.g. Alice"
        value=""
      />
    </label>
    <label>
      Class
      <input
        placeholder="e.g. Wizard"
        value=""
      />
    </label>
    <label>
      Level (default 1)
      <input
        min="1"
        type="number"
        value="1"
      />
    </label>
    <button
      type="submit"
    >
      Create
    </button>
  </form>
</div>
`;
